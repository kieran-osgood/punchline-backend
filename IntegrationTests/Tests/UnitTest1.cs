using System.Threading.Tasks;
using GraphQL;
using HotChocolate;
using HotChocolate.Execution;
using IntegrationTests.Helpers;
using Snapshooter.Xunit;
using Xunit;

namespace IntegrationTests.Tests
{
    public class UnitTest1 : IClassFixture<CustomWebApplicationFactory<Startup>>
    {
        private readonly CustomWebApplicationFactory<Startup> _factory;
        public UnitTest1(CustomWebApplicationFactory<Startup> factory) => _factory = factory;

        [Fact]
        public async Task GetJokes()
        {
            var factory = _factory.WithAuthentication(TestClaimsProvider.WithAdminClaims());
            const string query = @"query Jokes {
                  jokes(jokeLength: SMALL) {
                    nodes {
                      id
                      body
                    }
                  }
                }";

            var executor = await factory.Services.GetRequestExecutorAsync();
            var request = QueryRequestBuilder
                .New()
                .AddAuthorisedUser()
                .SetQuery(query)
                .Create();
            var result = await executor.ExecuteAsync(request);

            Assert.Null(result.Errors);
            (await result.ToJsonAsync()).MatchSnapshot();
        }


        [Theory]
        [InlineData(@"query Jokes {
                  jokes {
                    nodes {
                      id
                      body
                    }
                  }
                }")]
        public async Task Unauthenticated_User_Request(string query)
        {
            var request =
                QueryRequestBuilder
                    .New()
                    .SetQuery(query)
                    .Create();

            var executor = await _factory.Services.GetRequestExecutorAsync();
            var result = await executor.ExecuteAsync(request);

            (await result.ToJsonAsync()).MatchSnapshot();
        }
    }
}