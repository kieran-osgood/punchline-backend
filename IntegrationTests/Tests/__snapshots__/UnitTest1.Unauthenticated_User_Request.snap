{
  "errors": [
    {
      "message": "The current user is not authorized to access this resource.",
      "locations": [
        {
          "line": 2,
          "column": 19
        }
      ],
      "path": [
        "jokes"
      ],
      "extensions": {
        "code": "AUTH_NOT_AUTHENTICATED"
      }
    }
  ],
  "data": {
    "jokes": null
  }
}
